### PR TITLE
[cmd/mdatagen] Handle string validators in generated config structs

### LINKED ISSUE #14807: [cmd/mdatagen] Implement string validation in generated Go config code
Part of #14809

Implement support for the JSON Schema string validation keywords in Go code generated by `mdatagen`.

### Scope

- **`minLength`**: String must have at least the specified number of characters (Unicode codepoints).
- **`maxLength`**: String must have at most the specified number of characters (Unicode codepoints).
- **`pattern`**: String must match the specified regular expression.

### Acceptance Criteria

- Generated `Validate()` methods enforce `minLength`, `maxLength`, and `pattern` constraints.
- Validation errors clearly indicate the constraint that was violated.
- Unit tests cover edge cases (empty strings, boundary lengths, matching/non-matching patterns).

### PR DESCRIPTION
**Pull request by jkoronaAtCisco**
#### Description

Extends the mdatagen config generation pipeline to support string field validators defined in JSON Schema. When a config property specifies `minLength`, `maxLength`, or `pattern`, the generated `Validate()` method now emits the corresponding checks.

#### Link to tracking issue
Fixes #14807

#### Testing

- New unit tests in generation_test.go
- The samplescraper golden file (`generated_config.go`) is updated to reflect the new generated output.

### CODE REVIEW

File: cmd/mdatagen/internal/cfggen/generation.go [hunk 1]
```diff
@@ -312,25 +324,57 @@ func ExtractValidators(md *ConfigMetadata) []Validator {
 type Validator struct {
 	FieldName       string
 	FieldType       string
-	IsRequired      bool
 	IsPointer       bool
 	IsOptional      bool
+	Validators      map[string]any
```
Comment (dmitryax) [inline cmd/mdatagen/internal/cfggen/generation.go]: Why does it have to be a map here if we deal with predefined set of keys? If we use a struct here, it should simplify the code significantly if don't miss anything

Comment (jkoronaAtCisco) [reply cmd/mdatagen/internal/cfggen/generation.go]: That map represents validations (and their parameters) that are actually assigned to the field. Key is validation name and value are parameters, e.g.`minLength: 6`. In template I can simply check for key presence in the map and generate given validator. If map is empty there are no validators assigned to the field.

Comment (dmitryax) [reply cmd/mdatagen/internal/cfggen/generation.go]: Why can't it be?
```
type Validators struct {
	MaxLength *int
	MinLength *int
}
```
if a field is nil, it would mean no validation for that field

Comment (jkoronaAtCisco) [reply cmd/mdatagen/internal/cfggen/generation.go]: I followed your suggestion and refactored code accordingly. PTAL

File: cmd/mdatagen/internal/cfggen/generation.go [hunk 2]
```diff
@@ -309,28 +313,56 @@ func ExtractValidators(md *ConfigMetadata) []Validator {
 	return validators
 }
 
+type ValidationRules struct {
+	MaxLength *int
+	MinLength *int
+	Pattern   *string
+	Required  *bool
```
Comment (dmitryax) [inline cmd/mdatagen/internal/cfggen/generation.go]: I don't think we need the third state for required
```suggestion
	Required  bool
```

Comment (jkoronaAtCisco) [reply cmd/mdatagen/internal/cfggen/generation.go]: Good point, fixed.

**Review by dmitryax — APPROVED** [review]

File: cmd/mdatagen/internal/samplescraper/generated_config.go [hunk 3]
```diff
@@ -28,24 +33,36 @@ func (c *TargetsItem) Validate() error {
...
+	}
+	if len(c.JobName) > 255 {
+		err = errors.Join(err, errors.New("job_name exceeds maximum length of 255"))
+	}
+	if len(c.JobName) < 1 {
+		err = errors.Join(err, errors.New("job_name must have minimum length of 1"))
+	}
+	if matched, _ := regexp.MatchString(`^[a-zA-Z0-9_.-]+$`, c.JobName); !matched {
+		err = errors.Join(err, errors.New("job_name must match pattern `^[a-zA-Z0-9_.-]+$`"))
+	}
```
Comment (dmitryax) [inline cmd/mdatagen/internal/samplescraper/generated_config.go:76]: Let's generate tests for Validate() as well. Look at the existing patterns

Comment (jkoronaAtCisco) [reply cmd/mdatagen/internal/samplescraper/generated_config.go:76]: Tests are now generated. Scope:
- `createDefaultConfig` and `NewDefaultXXX` do not produce `nil`
- default object passes validation
- required validation triggers for empty fields

Example: https://github.com/open-telemetry/opentelemetry-collector/pull/15078/changes#diff-ac45afb26228a1ee02049ce963fdb643dfb6d828431aaef36141efb9c77a44fa
